From 1fdb6bf8ba28f66409fbcae78e15556f45d43623 Mon Sep 17 00:00:00 2001
From: Leorize <alaviss@users.noreply.github.com>
Date: Sun, 7 Jan 2018 09:22:55 +0700
Subject: CMakeLists: use check_symbol_exists()

CMake manual suggests using this module instead of CheckFunctionExists.
Also, this fixes iconv detection on OSes that use an external iconv
implementation (libiconv) as iconv is defined as a macro, which
CheckFunctionExists can't detect.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 39ae378..85915d0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,7 @@ set(PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}")
 option(BUILD_STATIC "Build static version of libunshield" OFF)
 
 include(CheckIncludeFiles)
-include(CheckFunctionExists)
+include(CheckSymbolExists)
 include(CheckCSourceCompiles)
 include(GNUInstallDirs)
 
@@ -26,8 +26,8 @@ check_include_files(sys/byteswap.h HAVE_SYS_BYTESWAP_H)
 check_include_files(sys/stat.h HAVE_SYS_STAT_H)
 check_include_files(sys/types.h HAVE_SYS_TYPES_H)
 check_include_files(unistd.h HAVE_UNISTD_H)
-check_function_exists(fnmatch HAVE_FNMATCH)
-check_function_exists(iconv HAVE_ICONV)
+check_symbol_exists(fnmatch fnmatch.h HAVE_FNMATCH)
+check_symbol_exists(iconv iconv.h HAVE_ICONV)
 
 set(SIZE_FORMAT "zi")
 check_c_source_compiles("#include <stdio.h>\nint main(int argc, char **argv) { size_t value = 0; printf(\"%${SIZE_FORMAT}\", value); return 0; }" SIZE_FORMAT_ZI)
-- 
2.15.0

